### PR TITLE
[IMP] utm: ignore capital letters

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
Task-3329888

### CODE REVIEW

File: addons/utm/models/utm_mixin.py [hunk 1]
```diff
@@ -59,6 +59,14 @@ def tracking_fields(self):
             ('utm_medium', 'medium_id', 'odoo_utm_medium'),
         ]
 
+    def _tracking_models(self):
```
Comment (awa-odoo) [inline addons/utm/models/utm_mixin.py:62]: Seems slightly overkill, but we could keep it I guess.

File: addons/utm/models/utm_mixin.py [hunk 2]
```diff
@@ -74,6 +82,17 @@ def _find_or_create_record(self, model_name, name):
...
+    def find_or_create_record(self, model_name, name):
+        """ Version of ``_find_or_create_record`` used in frontend notably in
+        website_links. For UTM models it calls _find_or_create_record. For other
+        models (as through inheritance custom models could be used) it simply
+        calls a create.
+
+        :return: id of newly created or found record """
+        if model_name in self._tracking_models():
+            return self._find_or_create_record(model_name, name).id
+        return self.env[model_name].create({self.env[model_name]._rec_name: name}).id
```
Comment (awa-odoo) [inline addons/utm/models/utm_mixin.py:94]: Why do we return the "id" specifically?

It's a little unusual and if compared to before, the JS call expected the raw return from "create".

Maybe I missed something.

File: addons/website_links/static/src/js/website_links.js [hunk 3]
```diff
@@ -132,13 +132,7 @@ var SelectBox = publicWidget.Widget.extend({
      * @param {String} name
      */
     _createObject: function (name) {
-        var args = {
-            name: name
-        };
-        if (this.obj === "utm.campaign") {
-            args.is_auto_campaign = true;
```
Comment (awa-odoo) [inline addons/website_links/static/src/js/website_links.js:139]: RIP?

**Review by awa-odoo — COMMENTED** [review]
Bloups

File: addons/utm/tests/test_utm.py [hunk 4]
```diff
@@ -57,6 +57,23 @@ def test_find_or_create_record(self):
...
+        source = self.env["utm.source"].create({
+            "name": name,
+        })
+        # case insensitive equal
+        for src in ("linkedin plus", "Linkedin plus", "LINKEDIN PLUS"):
+            with self.subTest(src=src):
+                found = self.env['utm.mixin']._find_or_create_record("utm.source", src)
+                self.assertEqual(found, source)
+        # not equal, just to be sure we don't do a pure ilike
+        for src in (f"{name} ", f" {name}", "LinkedIn", "Plus"):
```
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/utm/tests/test_utm.py]: Hagrid.

File: addons/utm/models/utm_mixin.py [hunk 5]
```diff
@@ -74,6 +82,17 @@ def _find_or_create_record(self, model_name, name):
 
         return record
 
+    def find_or_create_record(self, model_name, name):
```
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/utm/models/utm_mixin.py]: Ok.

File: addons/utm/models/utm_mixin.py (hunk 2 above)
Comment (tde-banana-odoo) [reply addons/utm/models/utm_mixin.py]: JS calls call_kw when using the "orm", and for create call_kw explicitly change the return type from a recordset to an id. As it is now a classic method, we have to manually return an ID. And api.returns does not work as model itself is dynamic.

File: addons/website_links/static/src/js/website_links.js [hunk 6]
```diff
@@ -132,13 +132,7 @@ var SelectBox = publicWidget.Widget.extend({
...
-        }
-        return this.orm.create(this.obj, [args]).then(record => {
+        return this.orm.call("utm.mixin", "find_or_create_record", [[], this.obj, name]).then(record => {
             this.$el.attr('value', record);
         });
```
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/website_links/static/src/js/website_links.js:137]: I don't like fancy things.

**Review by tde-banana-odoo — COMMENTED** [review]
Answered / updated

**Comment by awa-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**